### DISCUSSION TITLE
H2 database : JdbcSQLNonTransientConnectionException: Version mismatch

### DISCUSSION THREAD

**Discussion by Donadominic** [apache/skywalking#10291] [Q&A]
I want to view the data stored in h2 database.
So I changed h2 database mode to automatic mixed mode :
jdbcUrl: ${SW_STORAGE_H2_URL:jdbc:h2:/test;AUTO_SERVER=TRUE;DB_CLOSE_DELAY=-1}
 
 And it is listening to a port. But when I open the url localhost:port it is showing this error:

90047KVersion mismatch, driver version is "1195725856" but server version is "20"▒▒▒▒_▒▒org.h2.jdbc.JdbcSQLNonTransientConnectionException: Version mismatch, driver version is "1195725856" but server version is "20" [90047-212]
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:678)
        at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
        at org.h2.message.DbException.get(DbException.java:223)
        at org.h2.server.TcpServerThread.run(TcpServerThread.java:107)
        at java.lang.Thread.run(Thread.java:750)

h2-2.1.212.jar is the h2 jar file in oap-libs directory.
How to solve this problem?

**Comment by wu-sheng** [comment]
We can't know a H2 issue. This seems out of project scope.

**Reply by wu-sheng** [reply]
If you want a H2 server, usually boot one and listen, then use the OAP connecting it.

**Reply by Donadominic** [reply]
I installed h2 and configured its url in application.yaml . But when I query for data it is showing 0 rows . Why so?
![h2 console](https://user-images.githubusercontent.com/43172188/213146071-6ae1dd5b-3841-46f3-a387-cff480d92c02.png)

**Reply by wu-sheng** [reply]
This is not how we discuss on open source. 
You are setting env locally and private, but ask us why... how we can tell.
The tables are there, it means database connected. If there is no data, there is no telemetry data received.
